Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1642,6 +1634,20 @@ extension BrowserViewController: TabToolbarDelegate {
         guard let tab = tabManager.selectedTab else { return }
         HomePageHelper(prefs: profile.prefs).openHomePage(inTab: tab, withNavigationController: navigationController)
     }
+    
+    func showBackForwardList() {
+        if let backForwardList = tabManager.selectedTab?.webView?.backForwardList {
+            let backForwardViewController = BackForwardListViewController(profile: profile, backForwardList: backForwardList, isPrivate: tabManager.selectedTab?.isPrivate ?? false)
+            backForwardViewController.tabManager = tabManager
+            addChildViewController(backForwardViewController)
+            self.view.addSubview(backForwardViewController.view)
+            backForwardViewController.view.snp_makeConstraints { make in
+                make.top.equalTo(webViewContainerToolbar.snp_bottom)
+                make.left.right.bottom.equalTo(self.webViewContainer)
+                make.left.right.bottom.equalTo(self.webViewContainer)
+            }
+        }
```

Comment:
Instead of adding the view/view controller directly, it might be better to present the back forward list in a similar way we present the menu by _presenting_ the view controller instead of the view. We don't need the fancy animations the menu has but by implementing a custom `UIViewControllerAnimatedTransitioning` we can store the animation specific code for animating in/out the back forward list separately from the view controller. The `MenuPresentationAnimator` and `BrowserTrayAnimator` classes are more complex versions of this but you can use them as a reference for how we show/hide those components.